src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -1,2961 +0,0 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
-using System.Collections.Generic;
-using System.Collections.Immutable;
-using System.Diagnostics;
-using System.Linq;
-using System.Text;
-using Microsoft.CodeAnalysis.Collections;
-using Microsoft.CodeAnalysis.CSharp.Symbols;
-using Microsoft.CodeAnalysis.CSharp.Syntax;
-using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.CSharp
-{
-    internal abstract partial class PreciseAbstractFlowPass<LocalState> : BoundTreeVisitor
-        where LocalState : AbstractFlowPass<LocalState>.AbstractLocalState
-    {
-        protected int _recursionDepth;
-
-        /// <summary>
-        /// The compilation in which the analysis is taking place.  This is needed to determine which
-        /// conditional methods will be compiled and which will be omitted.
-        /// </summary>
-        protected readonly CSharpCompilation compilation;
```

Comment:
man is it unfortunate that in this PR this is showing up as a full delete/add of huge swaths of code.  Can you use partial classes to make it so that the PR diffs comes out cleaner?  Then, once revews, you can actually merge the partial parts together.

Reply:
I was expecting that was what he would do, but there are a number of systematic changes too (e.g. type parameter rename, loss of `_trackExceptions`, overridden methods)

---
In reply to: [236574390](https://github.com/dotnet/roslyn/pull/31389#discussion_r236574390) [](ancestors = 236574390)

Reply:
I tried to make this easier by separating each change into a separate commit. If you look at the by-commit view, does that make it simpler?

Reply:
sorry.  my bad.  i totally missed the "Note: I recommend reviewing commit-by-commit."  Sorry for wasting so much time :-/   I'll read things more carefully next time around!

Reply:
@CyrusNajmabadi No, I added that after you started your review, I forgot to mention that in the description -- no trouble at all!